lib/flight_scheduler/configuration.rb

```diff
@@ -0,0 +1,86 @@
+#==============================================================================
+# Copyright (C) 2020-present Alces Flight Ltd.
+#
+# This file is part of FlightSchedulerController.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# FlightSchedulerController is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with FlightSchedulerController. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on FlightSchedulerController, please visit:
+# https://github.com/openflighthpc/flight-scheduler-controller
+#==============================================================================
+require 'async'
+
+require_relative '../../app/models/node'
+require_relative '../../app/models/partition'
+
+module FlightScheduler
+  class Configuration
+    autoload(:Loader, 'flight_scheduler/configuration/loader')
+
+    ATTRIBUTES = [
+      {
+        name: :bind_address,
+        env_var: true,
+        default: 'http://127.0.0.1:6307',
```

Comment:
This doesn't look quite right as a default. As this is hosted on `127.0.0.1` it isn't accessible from other nodes, so it will need to sit behind a reverse proxy. However `6307` isn't a privileged port which opens the possibility of a MITM attack by non-root users.

There are three options to resolve this:
* Continue using the development port on `http://0.0.0.0:6307` without a reverse proxy,
* Use a reverse proxy to a privileged port: `http://127.0.0.1:918`, or
* Use both: `http://0.0.0.0:918`

Reply:
This default value is suitable for development.  It doesn't expose the controller on all interfaces and doesn't bind to a privileged port.

For production, we'd likely only want to expose the controller on the internal cluster network, not all networks.  As we can't determine what that is, the sysadmin installing flight scheduler will have to configure that.

The sysadmin is also going to have to configure the partitions and quite possibly other things too, so I don't see this as a big issue.